src/main/java/forceitembattle/listener/Listeners.java

```diff
@@ -65,42 +71,117 @@ public void onPlayerQuit(PlayerQuitEvent playerQuitEvent) {
         playerQuitEvent.getPlayer().getPassengers().forEach(Entity::remove);
     }
 
+    /* Found-/Skip Item */
+    @EventHandler
+    public void onPickupEvent(EntityPickupItemEvent entityPickupItemEvent) {
+        if(entityPickupItemEvent.getEntity() instanceof Player player) {
+            if(ForceItemBattle.getTimer().isRunning()) {
+                ItemStack pickedItem = entityPickupItemEvent.getItem().getItemStack();
+                Material currentMaterial = ForceItemBattle.getGamemanager().getCurrentMaterial(player);
+
+                if(pickedItem.getType() == currentMaterial) {
+                    FoundItemEvent foundItemEvent = new FoundItemEvent(player);
+                    foundItemEvent.setFoundItem(pickedItem);
+                    foundItemEvent.skipped(false);
+
+                    Bukkit.getPluginManager().callEvent(foundItemEvent);
+                }
+            }
+        }
+    }
+
+    @EventHandler
+    public void onFoundItemInInventory(InventoryClickEvent inventoryClickEvent) {

```

Comment:
Make sure that the item actually is in the player's inventory not simply if you click the item. Otherwise for exaple with the /info inventory it will also trigger if you click the item.

Reply:
There are also a lot of other events that can cause the content of a players inventory to change (Interacting with armor stand, Dispenser immediately dropping armor into your inventory, ...). I guess it really is simpler to just check the players inventory for the specific item every tick. It might sound "inefficient" but it should not cause any lag whatsoever. Maybe time the checker thread or something... Should really not be more than 1ms for checking all players inventories.

Reply:
The /info should not be the problem, you could simply by cancel the inventory to prevent this. In addition, the FoundItem is only called after you have clicked on the item, what you need. Don't see that you could - apart from the /info yet - click on the required item from somewhere else without being able to take it out already

Reply:
At the armor stand you can easily do that with the InteractEntity event or so. The dispenser does not matter i guess, as ItemPickupEvent is already handled